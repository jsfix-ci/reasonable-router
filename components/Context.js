import createContext from 'create-react-context';

// eslint-disable-next-line no-unused-vars
const push = (path, state = {}) => {};
const routerRenderProperties = {
	params: {},
	Component: () => null,
};
const currentLocation = {
	hash: '',
	pathname: '/',
	search: '',
	state: undefined,
};
const routes = {};

const defaultContext = {
	push,
	routerRenderProperties,
	currentLocation,
	routes,
};

export const Context = createContext(defaultContext);
