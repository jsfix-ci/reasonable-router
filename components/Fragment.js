import React from 'react';
import PropTypes from 'prop-types';
import { matchSingleRoute } from './matchRoute';


const Fragment = (props, context) => {
	const { getCurrentLocation } = context;
	const { forRoute, children } = props;

	const route = matchSingleRoute(forRoute, getCurrentLocation().pathname);

	if (!route) {
		return null;
	}

	const childrenIsReactElement =
		children &&
		children.type && typeof children.type === 'function' &&
		children.$$typeof && children.$$typeof === Symbol.for('react.element');

	const childrenWithParams = childrenIsReactElement
		? React.cloneElement(children, { params: route.params } )
		: children;

	return React.Children.only(childrenWithParams);
};
Fragment.contextTypes = {
	getCurrentLocation: PropTypes.func,
};

export default Fragment;
