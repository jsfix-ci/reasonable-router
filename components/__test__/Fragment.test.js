import React from 'react';
import { render, shallow } from 'enzyme';

import { Fragment } from '../Fragment';
import { Context } from '../Context';

describe('<Fragment />', () => {
	const location = (pathname) => ({
		currentLocation: { pathname },
	});

	it('renders when the route matches', () => {
		const result = shallow(
			<Context.Provider value={location('/page')}>
				<Fragment forRoute="/page">
					<h1>Text</h1>
				</Fragment>
			</Context.Provider>,
		);

		expect(result.contains(<h1>Text</h1>)).toBe(true);
		expect(result).toMatchSnapshot();
	});

	it('does not render when the route does not match', () => {
		const result = render(
			<Context.Provider value={location('/page')}>
				<Fragment forRoute="/another-page">
					<h1>Text</h1>
				</Fragment>
			</Context.Provider>,
		);

		expect(result.find('h1')).toHaveLength(0);
		expect(result).toMatchSnapshot();
	});

	it('passes route params as props', () => {
		const PropPrinter = ({ params }) => <p>{JSON.stringify(params)}</p>;
		const result = render(
			<Fragment forRoute="/:name">
				<PropPrinter />
			</Fragment>,
			{ context: location('/a-name') },
		)
			.first()
			.shallow(); // Render <PropPrinter /> as well

		expect(result.text()).toEqual(JSON.stringify({ name: 'a-name' }));
	});
});
