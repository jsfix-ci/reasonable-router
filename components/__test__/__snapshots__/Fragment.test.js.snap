// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fragment /> does not render when the route does not match 1`] = `
initialize {
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`<Fragment /> renders when the route matches 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    value={
      Object {
        "currentLocation": Object {
          "pathname": "/page",
        },
      }
    }
  >
    <Fragment
      forRoute="/page"
    >
      <h1>
        Text
      </h1>
    </Fragment>
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <h1>
        Text
      </h1>,
      "forRoute": "/page",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Text",
      },
      "ref": null,
      "rendered": "Text",
      "type": "h1",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <h1>
          Text
        </h1>,
        "forRoute": "/page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Text",
        },
        "ref": null,
        "rendered": "Text",
        "type": "h1",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
